presto-hive/src/main/java/com/facebook/presto/hive/HiveColumnHandle.java

```diff
@@ -153,4 +164,21 @@ public static HiveColumnHandle updateRowIdHandle(String connectorId)
         // the plan-time support needed just works.
         return new HiveColumnHandle(connectorId, UPDATE_ROW_ID_COLUMN_NAME, HIVE_LONG, BIGINT.getTypeSignature(), -1, true);
     }
+
+    public static HiveColumnHandle pathColumnHandle(String connectorId)
+    {
+        // Like partition columns, this column can not be found in data source. Using true for partitionKey,
+        // the plan-time support needed just works.
+        return new HiveColumnHandle(connectorId, PATH_COLUMN_NAME, PATH_HIVE_TYPE, PATH_TYPE_SIGNATURE, PATH_COLUMN_INDEX, true);
```

Comment:
```
    public static HiveColumnHandle updateRowIdHandle(String connectorId)
    {
        // Hive connector only supports metadata delete. It does not support generic row-by-row deletion.
        // Metadata delete is implemented in Presto by generating a plan for row-by-row delete first,
        // and then optimize it into metadata delete. As a result, Hive connector must provide partial
        // plan-time support for row-by-row delete so that planning doesn't fail. This is why we need
        // rowid handle. Note that in Hive connector, rowid handle is not implemented beyond plan-time.

        // Like partition columns, this column can not be found in data source. Using true for partitionKey,
        // the plan-time support needed just works.
        return new HiveColumnHandle(connectorId, UPDATE_ROW_ID_COLUMN_NAME, HIVE_LONG, BIGINT.getTypeSignature(), -1, true);
    }
```